Marlin/Conditionals_post.h

```diff
@@ -731,7 +731,7 @@
    * Set granular options based on the specific type of leveling
    */
 
-  #define UBL_DELTA  (ENABLED(AUTO_BED_LEVELING_UBL) && ENABLED(DELTA))
+  #define UBL_DELTA  (ENABLED(AUTO_BED_LEVELING_UBL) && (ENABLED(DELTA) || ENABLED(UBL_DELTA_ON_CARTESIAN)))
```

Comment:
OK, I changed the name from UBL_DELTA_ON_CARTESIAN which is confusing to UBL_GRANULAR_SEGMENTATION_FOR_CARTESIAN to better reflect its meaning.  It is NOT the default for UBL on cartesian, just an option.